lib/time/time.go

```diff
@@ -0,0 +1,515 @@
+var Module = &starlarkstruct.Module{
+	Name: ModuleName,
+	Members: starlark.StringDict{
+		"duration":       starlark.NewBuiltin("duration", newDuration),
+		"parse_duration": starlark.NewBuiltin("parse_duration", parseDuration),
+		"parse_location": starlark.NewBuiltin("parse_location", parseLocation),
+		"now":            starlark.NewBuiltin("now", now),
+		"struct":         starlark.NewBuiltin("struct", starlarkstruct.Make),
+		"time":           starlark.NewBuiltin("time", newTime),
+		"parse_time":     starlark.NewBuiltin("time", parseTime),
+		"sleep":          starlark.NewBuiltin("sleep", sleep),
+		"from_timestamp": starlark.NewBuiltin("from_timestamp", fromTimestamp),
+
+		"zero": Time{},
+
+		"nanosecond":  Duration(time.Nanosecond),
+		"microsecond": Duration(time.Microsecond),
+		"millisecond": Duration(time.Millisecond),
+		"second":      Duration(time.Second),
+		"minute":      Duration(time.Minute),
+		"hour":        Duration(time.Hour),
+	},
+}
+
+// LoadModule loads the time module.
+// It is concurrency-safe and idempotent.
+func LoadModule() (starlark.StringDict, error) {
+	return starlark.StringDict{
+		ModuleName: Module,
+	}, nil
+}
+
+// NowFunc is a function that generates the current time. Intentionally exported
+// so that it can be overridden, for example by applications that require their
+// Starlark scripts to be fully deterministic.
+var NowFunc = time.Now
+
+func parseDuration(thread *starlark.Thread, _ *starlark.Builtin, args starlark.Tuple, kwargs []starlark.Tuple) (starlark.Value, error) {
+	var d Duration
+	err := starlark.UnpackArgs("duration", args, kwargs, "d", &d)
+	return d, err
+}
+
+func parseLocation(thread *starlark.Thread, _ *starlark.Builtin, args starlark.Tuple, kwargs []starlark.Tuple) (starlark.Value, error) {
+	var s string
+	if err := starlark.UnpackArgs("location", args, kwargs, "s", &s); err != nil {
+		return nil, err
+	}
+	loc, err := time.LoadLocation(s)
+	if err != nil {
+		return nil, err
+	}
+
+	return starlark.String(loc.String()), nil
+}
+
+// SleepFunc is a function that pauses the current goroutine for at least d.
+// Intentionally exported so that it can be overridden, for example by
+// applications that require their Starlark scripts to be fully deterministic.
+var SleepFunc = time.Sleep
+
+func sleep(thread *starlark.Thread, _ *starlark.Builtin, args starlark.Tuple, kwargs []starlark.Tuple) (starlark.Value, error) {
+	var dur Duration
+	if err := starlark.UnpackPositionalArgs("sleep", args, kwargs, 1, &dur); err != nil {
+		return starlark.None, err
+	}
+
+	SleepFunc(time.Duration(dur))
+	return starlark.None, nil
+}
+
+func parseTime(thread *starlark.Thread, _ *starlark.Builtin, args starlark.Tuple, kwargs []starlark.Tuple) (starlark.Value, error) {
+	var (
+		x, location string
+		format      = time.RFC3339
+	)
+	if err := starlark.UnpackArgs("time", args, kwargs, "x", &x, "format?", &format, "location", &location); err != nil {
```

Comment:
I think this should be
```suggestion
	if err := starlark.UnpackArgs("time", args, kwargs, "x", &x, "format?", &format, "location?", &location); err != nil {
```
to also make `location` optional, shouldn't it?

Reply:
great catch!